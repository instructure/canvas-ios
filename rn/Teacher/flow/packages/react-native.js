//
// This file is part of Canvas.
// Copyright (C) 2017-present  Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//

// @flow

declare var console: typeof console & { disableYellowBox: boolean }

export type PanResponderGestureState = {
  stateID: number,
  moveX: number,
  moveY: number,
  x0: number,
  y0: number,
  dx: number,
  dy: number,
  vx: number,
  vy: number,
}

export type KeyboardEventData = {
  endCoordinates: {
    width: number,
    height: number,
    screenX: number,
    screenY: number,
  },
}
