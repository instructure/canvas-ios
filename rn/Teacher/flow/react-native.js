//
// This file is part of Canvas.
// Copyright (C) 2017-present  Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//

// @flow
/* eslint-disable */

// define basic style prop types
// the internal ones are the best we have
import {
  type ____ViewStyleProp_Internal,
  type ____TextStyleProp_Internal,
  type ____ImageStyleProp_Internal,
} from 'react-native/Libraries/StyleSheet/StyleSheetTypes'

declare type ViewStyleProp = ____ViewStyleProp_Internal
declare type TextStyleProp = ____TextStyleProp_Internal
declare type ImageStyleProp = ____ImageStyleProp_Internal
