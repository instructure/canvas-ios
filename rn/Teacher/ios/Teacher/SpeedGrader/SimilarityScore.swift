//
// This file is part of Canvas.
// Copyright (C) 2020-present  Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//

import SwiftUI
import Core

struct SimilarityScore: View {
    let status: String
    let score: Double
    let url: URL?

    init?(_ submission: Submission, file: File?) {
        if submission.type == .online_upload, let file = file, let status = file.similarityStatus {
            self.status = status
            self.score = file.similarityScore
            self.url = file.similarityURL
        } else if submission.type == .online_text_entry, let status = submission.similarityStatus {
            self.status = status
            self.score = submission.similarityScore
            self.url = submission.similarityURL
        } else {
            return nil
        }
    }

    var body: some View {
        let content = HStack(spacing: 0) {
            Text("Similarity Score")
                .font(.semibold14).foregroundColor(.textDarkest)
            Spacer()
            switch status {
            case "scored":
                Text(NumberFormatter.localizedString(from: NSNumber(value: score / 100), number: .percent))
                    .font(.semibold16).foregroundColor(.white)
                    .padding(EdgeInsets(top: 4, leading: 8, bottom: 4, trailing: 8))
                    .background(
                        score >= 75 ? Color.textDanger :
                        score >= 50 ? Color.textAlert :
                        score >= 25 ? Color.textWarning :
                        score >= 1 ? Color.textSuccess :
                        Color.textInfo
                    )
                    .cornerRadius(4)
            case "pending":
                Image.clockLine
                    .size(18).foregroundColor(.white)
                    .padding(4).background(Color.backgroundDark).cornerRadius(4)
                    .accessibility(label: Text("Pending"))
            default:
                Image.warningLine
                    .size(18).foregroundColor(.white)
                    .padding(4).background(Color.backgroundDanger).cornerRadius(4)
                    .accessibility(label: Text("Error"))
            }
            if url != nil {
                Spacer().frame(width: 8)
                DisclosureIndicator()
            }
        }
            .padding(EdgeInsets(top: 4, leading: 16, bottom: 4, trailing: 16))
        VStack(spacing: 0) {
            if let url = url {
                Button(action: { Router.open(url: .parse(url)) }, label: { content })
            } else {
                content
            }
            Divider()
        }
    }
}
