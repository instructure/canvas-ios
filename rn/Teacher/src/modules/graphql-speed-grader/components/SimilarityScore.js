//
// This file is part of Canvas.
// Copyright (C) 2017-present  Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//

// @flow

import React, { Component } from 'react'
import i18n from 'format-message'
import { colors, createStyleSheet } from '../../../common/stylesheet'
import images from '../../../images'
import DisclosureIndicator from '../../../common/components/DisclosureIndicator'

import {
  Text,
  TouchableHighlight,
  View,
  Image,
} from 'react-native'

type StateProps = {
  status: ?string,
  score: ?number,
  url: ?string,
}

export type Props = StateProps

export default class SimilarityScore extends Component<Props, any> {
  render () {
    let selectedSubmission = this.props.submission.submissionHistory.edges[this.props.selectedIndex].submission
    if (selectedSubmission.turnitinData == null) return null

    if (!['online_text_entry', 'online_upload'].includes(selectedSubmission.submissionType)) return null

    let turnitinData
    if (selectedSubmission.submissionType === 'online_text_entry') {
      turnitinData = selectedSubmission.turnitinData
        .find(data => data.target.__typename === 'Submission' && data.target._id === selectedSubmission.rootId)
    } else if (selectedSubmission.submissionType === 'online_upload') {
      let selectedAttachmentIndex = this.props.selectedAttachmentIndex
      let selectedAttachment = selectedSubmission.attachments[selectedAttachmentIndex]
      turnitinData = selectedSubmission.turnitinData
        .find(data => data.target.__typename === 'File' && data.target._id === selectedAttachment._id)
    }

    if (turnitinData == null) return null

    const { status, score } = turnitinData
    if (!status) {
      return null
    }

    return (
      <TouchableHighlight
        underlayColor='white'
        testID='speedgrader.similarity-score.container'
      >
        <View style={style.container}>
          <Text
            style={style.label}
            testID='speedgrader.similarity-score.label'
          >
            {i18n('Similarity Score')}
          </Text>
          { status === 'scored' && score != null && this.renderScore(score) }
          { status === 'pending' && this.renderPending() }
          { status !== 'scored' && status !== 'pending' && this.renderError() }
        </View>
      </TouchableHighlight>
    )
  }

  renderScore (score: number) {
    let color = colors.textInfo
    switch (true) {
      case (score >= 75):
        color = colors.textDanger
        break
      case (score >= 50):
        color = colors.textAlert
        break
      case (score >= 25):
        color = colors.textWarning
        break
      case (score >= 1):
        color = colors.textSuccess
        break
      default:
        color = colors.textInfo
    }

    return (
      <View style={{ flexDirection: 'row' }}>
        <View
          style={[style.scoreContainer, { backgroundColor: color }]}
          testID='speedgrader.similarity-score.score.container'
        >
          <Text
            style={style.score}
            testID='speedgrader.similarity-score.score.label'
          >
            {`${score}%`}
          </Text>
        </View>
        { Boolean(this.props.url) &&
          <DisclosureIndicator />
        }
      </View>
    )
  }

  renderPending () {
    return this.renderStatus(images.speedGrader.turnitin.pending, '#72818B')
  }

  renderError () {
    return this.renderStatus(images.speedGrader.turnitin.error, '#F30017')
  }

  renderStatus (icon: any, color: string) {
    return (
      <View
        style={[style.statusContainer, { backgroundColor: color }]}
        testID='speedgrader.similarity-score.status.container'
      >
        <Image
          source={icon}
          style={style.statusIcon}
          testID='speedgrader.similarity-score.status.icon'
        />
      </View>
    )
  }
}

const style = createStyleSheet((colors, vars) => ({
  container: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    borderBottomColor: '#D8D8D8',
    borderBottomWidth: vars.hairlineWidth,
    borderStyle: 'solid',
    marginHorizontal: vars.padding,
    paddingVertical: 4,
    alignItems: 'center',
    backgroundColor: colors.white,
  },
  label: {
    fontSize: 14,
    color: colors.textDark,
    fontFamily: '.SFUIDisplay-semibold',
  },
  scoreContainer: {
    paddingHorizontal: 8,
    paddingVertical: 4,
    borderRadius: 4,
  },
  score: {
    fontSize: 16,
    color: colors.white,
    fontFamily: '.SFUIDisplay-semibold',
  },
  statusIcon: {
    width: 18,
    height: 18,
    tintColor: colors.white,
  },
  statusContainer: {
    borderRadius: 4,
    padding: 4,
  },
}))
