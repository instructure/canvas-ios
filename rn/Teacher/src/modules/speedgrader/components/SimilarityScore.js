//
// This file is part of Canvas.
// Copyright (C) 2017-present  Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//

// @flow

import React, { Component } from 'react'
import { connect } from 'react-redux'
import i18n from 'format-message'
import { colors, createStyleSheet } from '../../../common/stylesheet'
import images from '../../../images'
import DisclosureIndicator from '../../../common/components/DisclosureIndicator'

import {
  Text,
  TouchableHighlight,
  View,
  Image,
} from 'react-native'

type OwnProps = {
  submissionID: ?string,
}

type StateProps = {
  status: ?string,
  score: ?number,
  url: ?string,
}

export type Props = StateProps

export class SimilarityScore extends Component<Props, any> {
  render () {
    const { status, score } = this.props
    if (!status) {
      return null
    }

    return (
      <TouchableHighlight
        underlayColor={colors.backgroundLightest}
        testID='speedgrader.similarity-score.container'
      >
        <View style={style.container}>
          <Text
            style={style.label}
            testID='speedgrader.similarity-score.label'
          >
            {i18n('Similarity Score')}
          </Text>
          { status === 'scored' && score != null && this.renderScore(score) }
          { status === 'pending' && this.renderPending() }
          { status !== 'scored' && status !== 'pending' && this.renderError() }
        </View>
      </TouchableHighlight>
    )
  }

  renderScore (score: number) {
    let color = colors.textInfo
    switch (true) {
      case (score >= 75):
        color = colors.textDanger
        break
      case (score >= 50):
        color = colors.textAlert
        break
      case (score >= 25):
        color = colors.textWarning
        break
      case (score >= 1):
        color = colors.textSuccess
        break
      default:
        color = colors.textInfo
    }

    return (
      <View style={{ flexDirection: 'row' }}>
        <View
          style={[style.scoreContainer, { backgroundColor: color }]}
          testID='speedgrader.similarity-score.score.container'
        >
          <Text
            style={style.score}
            testID='speedgrader.similarity-score.score.label'
          >
            {`${score}%`}
          </Text>
        </View>
        { Boolean(this.props.url) &&
          <DisclosureIndicator />
        }
      </View>
    )
  }

  renderPending () {
    return this.renderStatus(images.speedGrader.turnitin.pending, colors.textDark)
  }

  renderError () {
    return this.renderStatus(images.speedGrader.turnitin.error, colors.textDanger)
  }

  renderStatus (icon: any, color: string) {
    return (
      <View
        style={[style.statusContainer, { backgroundColor: color }]}
        testID='speedgrader.similarity-score.status.container'
      >
        <Image
          source={icon}
          style={style.statusIcon}
          testID='speedgrader.similarity-score.status.icon'
        />
      </View>
    )
  }
}

const style = createStyleSheet((colors, vars) => ({
  container: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    borderBottomColor: colors.borderMedium,
    borderBottomWidth: vars.hairlineWidth,
    borderStyle: 'solid',
    marginHorizontal: 16,
    paddingVertical: 4,
    alignItems: 'center',
    backgroundColor: colors.backgroundLightest,
  },
  label: {
    fontSize: 14,
    color: colors.textDarkest,
    fontWeight: '600',
  },
  scoreContainer: {
    paddingHorizontal: 8,
    paddingVertical: 4,
    borderRadius: 4,
  },
  score: {
    fontSize: 16,
    color: colors.white,
    fontWeight: '600',
  },
  statusIcon: {
    width: 18,
    height: 18,
    tintColor: colors.white,
  },
  statusContainer: {
    borderRadius: 4,
    padding: 4,
  },
}))

export function mapStateToProps (state: AppState, ownProps: OwnProps): StateProps {
  const { submissionID } = ownProps

  if (submissionID &&
    state.entities.submissions[submissionID] &&
    state.entities.submissions[submissionID].submission) {
    // get selected submission
    const submissionData = state.entities.submissions[submissionID]
    const { submission, selectedIndex, selectedAttachmentIndex } = submissionData
    const selectedSubmission = selectedIndex != null ? submission.submission_history[selectedIndex] : submission

    if (selectedSubmission && selectedSubmission.turnitin_data) {
      // online text entry
      if (selectedSubmission.submission_type === 'online_text_entry') {
        const data = selectedSubmission.turnitin_data[`submission_${selectedSubmission.id}`]
        if (data) {
          return mapTurnItInDataToProps(data)
        }
      }

      // online upload
      if (selectedSubmission.submission_type === 'online_upload') {
        if (selectedSubmission.attachments && selectedSubmission.attachments.length > 0) {
          const { attachments } = selectedSubmission
          const selectedAttachment = selectedAttachmentIndex != null ? attachments[selectedAttachmentIndex] : attachments[0]
          const data = selectedSubmission.turnitin_data[`attachment_${selectedAttachment.id}`]
          if (data) {
            return mapTurnItInDataToProps(data)
          }
        }
      }
    }
  }

  return {
    status: null,
    score: null,
    url: null,
  }
}

function mapTurnItInDataToProps (data: TurnItInData): StateProps {
  const { status, similarity_score } = data
  const url = data.outcome_response ? data.outcome_response.outcomes_tool_placement_url : null
  return {
    status,
    url,
    score: similarity_score,
  }
}

let Connected = connect(mapStateToProps)(SimilarityScore)
export default (Connected: any)
