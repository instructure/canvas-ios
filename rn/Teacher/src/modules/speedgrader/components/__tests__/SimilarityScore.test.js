//
// This file is part of Canvas.
// Copyright (C) 2017-present  Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//

/* @flow */

import React from 'react'
import 'react-native'
import renderer from 'react-test-renderer'
import { SimilarityScore, type Props, mapStateToProps } from '../SimilarityScore'
import explore from '../../../../../test/helpers/explore'

jest
  .mock('react-native/Libraries/Components/Button', () => 'Button')
  .mock('react-native/Libraries/Components/Touchable/TouchableHighlight', () => 'TouchableHighlight')
  .mock('react-native/Libraries/Components/Touchable/TouchableOpacity', () => 'TouchableOpacity')

const template = {
  ...require('../../../../__templates__/submissions'),
  ...require('../../../../redux/__templates__/app-state'),
}

describe('SimilarityScore', () => {
  let props: Props
  beforeEach(() => {
    props = {
      score: 0,
      status: 'scored',
      url: null,
    }
  })

  it('displays label', () => {
    const view = render(props)
    const label: any = explore(view.toJSON()).selectByID('speedgrader.similarity-score.label')
    expect(label.children).toContain('Similarity Score')
  })

  it('displays score', () => {
    props.score = 20
    const score: any = explore(render(props).toJSON()).selectByID('speedgrader.similarity-score.score.label')
    expect(score.children).toContain('20%')
  })

  it('uses the correct color depending on the score', () => {
    testScoreColor(0)
    testScoreColor(1)
    testScoreColor(25)
    testScoreColor(50)
    testScoreColor(75)
    testScoreColor(100)
  })

  it('displays disclosure indicator if there is a url', () => {
    props.url = 'some url'
    expect(explore(render(props).toJSON()).selectByType('DisclosureIndicator')).not.toBeNull()
  })

  it('renders null without a status', () => {
    props.status = null
    expect(render(props).toJSON()).toBeNull()
  })

  it('renders pending status', () => {
    props.status = 'pending'
    const status: any = explore(render(props).toJSON()).selectByID('speedgrader.similarity-score.status.container')
    expect(status).toMatchSnapshot()
  })

  it('renders error status', () => {
    props.status = 'error'
    const status: any = explore(render(props).toJSON()).selectByID('speedgrader.similarity-score.status.container')
    expect(status).toMatchSnapshot()
  })

  function render (props: Props): any {
    return renderer.create(<SimilarityScore {...props} />)
  }

  function testScoreColor (score: number) {
    props.score = score
    const container: any = explore(render(props).toJSON()).selectByID('speedgrader.similarity-score.score.container')
    expect(container.props.style).toMatchSnapshot()
  }
})

describe('mapStateToProps', () => {
  it('maps text submission to props', () => {
    const state = template.appState({
      entities: {
        submissions: {
          '1': {
            submission: template.submissionHistory([{
              id: '1',
              submission_type: 'online_text_entry',
              turnitin_data: {
                submission_1: {
                  status: 'scored',
                  similarity_score: 20,
                  outcome_response: {
                    outcomes_tool_placement_url: 'https://some-site-at-turnitin.com',
                  },
                },
              },
            }]),
          },
        },
      },
    })
    expect(
      mapStateToProps(state, { submissionID: '1' })
    ).toEqual({
      score: 20,
      status: 'scored',
      url: 'https://some-site-at-turnitin.com',
    })
  })

  it('maps file submission to props', () => {
    const state = template.appState({
      entities: {
        submissions: {
          '1': {
            submission: template.submissionHistory([{
              id: '1',
              submission_type: 'online_upload',
              attachments: [{
                id: '2',
              }],
              turnitin_data: {
                attachment_2: {
                  status: 'scored',
                  similarity_score: 20,
                },
              },
            }]),
          },
        },
      },
    })
    expect(
      mapStateToProps(state, { submissionID: '1' })
    ).toEqual({
      score: 20,
      status: 'scored',
      url: null,
    })
  })

  it('maps selected submission and selected attachment to props', () => {
    const state = template.appState({
      entities: {
        submissions: {
          '2': {
            selectedIndex: 0,
            selectedAttachmentIndex: 1,
            submission: template.submissionHistory([
              {
                id: '1',
                submission_type: 'online_upload',
                attachments: [
                  {
                    id: '1',
                  },
                  {
                    id: '2',
                  },
                  {
                    id: '3',
                  },
                ],
                turnitin_data: {
                  attachment_1: {
                    status: 'scored',
                    similarity_score: 10,
                  },
                  attachment_2: {
                    status: 'scored',
                    similarity_score: 20,
                  },
                  attachment_3: {
                    status: 'scored',
                    similarity_score: 30,
                  },
                },
              },
              {
                id: '2',
              },
            ]),
          },
        },
      },
    })
    expect(
      mapStateToProps(state, { submissionID: '2' })
    ).toEqual({
      status: 'scored',
      score: 20,
      url: null,
    })
  })

  it('handles no submission id', () => {
    expect(
      mapStateToProps(template.appState(), { submissionID: null })
    ).toEqual({
      status: null,
      url: null,
      score: null,
    })
  })

  it('handles no turnitin data', () => {
    const state = template.appState({
      entities: {
        submissions: {
          '1': {
            submission: template.submissionHistory([{
              id: '1',
              turnitin_data: undefined,
            }]),
          },
        },
      },
    })
    expect(
      mapStateToProps(state, { submissionID: '1' })
    ).toEqual({
      status: null,
      url: null,
      score: null,
    })
  })

  it('handles submission types not supported by turnitin', () => {
    const state = template.appState({
      entities: {
        submissions: {
          '1': {
            submission: template.submissionHistory([{
              id: '1',
              submission_type: 'online_url',
              turnitin_data: {
                submission_1: {
                  status: 'scored',
                  similarity_score: 20,
                },
              },
            }]),
          },
        },
      },
    })
    expect(
      mapStateToProps(state, { submissionID: '1' })
    ).toEqual({
      status: null,
      url: null,
      score: null,
    })
  })

  it('handles uploads without attachments', () => {
    const state = template.appState({
      entities: {
        submissions: {
          '1': {
            submission: template.submissionHistory([{
              id: '1',
              submission_type: 'online_upload',
              attachments: [],
              turnitin_data: {
                attachment_1: {
                  status: 'scored',
                  similarity_score: 20,
                },
              },
            }]),
          },
        },
      },
    })
    expect(
      mapStateToProps(state, { submissionID: '1' })
    ).toEqual({
      status: null,
      url: null,
      score: null,
    })
  })
})
