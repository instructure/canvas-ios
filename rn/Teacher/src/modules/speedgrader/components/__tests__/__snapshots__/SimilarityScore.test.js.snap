// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimilarityScore renders error status 1`] = `
<View
  style={
    Array [
      Object {
        "borderRadius": 4,
        "padding": 4,
      },
      Object {
        "backgroundColor": "#EE0612",
      },
    ]
  }
  testID="speedgrader.similarity-score.status.container"
>
  <Image
    source={
      Object {
        "testUri": "../../../src/images/Warning.png",
      }
    }
    style={
      Object {
        "height": 18,
        "tintColor": "#FFFFFF",
        "width": 18,
      }
    }
    testID="speedgrader.similarity-score.status.icon"
  />
</View>
`;

exports[`SimilarityScore renders pending status 1`] = `
<View
  style={
    Array [
      Object {
        "borderRadius": 4,
        "padding": 4,
      },
      Object {
        "backgroundColor": "#8B969E",
      },
    ]
  }
  testID="speedgrader.similarity-score.status.container"
>
  <Image
    source={
      Object {
        "testUri": "../../../src/images/clock.png",
      }
    }
    style={
      Object {
        "height": 18,
        "tintColor": "#FFFFFF",
        "width": 18,
      }
    }
    testID="speedgrader.similarity-score.status.icon"
  />
</View>
`;

exports[`SimilarityScore uses the correct color depending on the score 1`] = `
Array [
  Object {
    "borderRadius": 4,
    "paddingHorizontal": 8,
    "paddingVertical": 4,
  },
  Object {
    "backgroundColor": "#008EE2",
  },
]
`;

exports[`SimilarityScore uses the correct color depending on the score 2`] = `
Array [
  Object {
    "borderRadius": 4,
    "paddingHorizontal": 8,
    "paddingVertical": 4,
  },
  Object {
    "backgroundColor": "#00AC18",
  },
]
`;

exports[`SimilarityScore uses the correct color depending on the score 3`] = `
Array [
  Object {
    "borderRadius": 4,
    "paddingHorizontal": 8,
    "paddingVertical": 4,
  },
  Object {
    "backgroundColor": "#FC5E13",
  },
]
`;

exports[`SimilarityScore uses the correct color depending on the score 4`] = `
Array [
  Object {
    "borderRadius": 4,
    "paddingHorizontal": 8,
    "paddingVertical": 4,
  },
  Object {
    "backgroundColor": "#BF32A4",
  },
]
`;

exports[`SimilarityScore uses the correct color depending on the score 5`] = `
Array [
  Object {
    "borderRadius": 4,
    "paddingHorizontal": 8,
    "paddingVertical": 4,
  },
  Object {
    "backgroundColor": "#EE0612",
  },
]
`;

exports[`SimilarityScore uses the correct color depending on the score 6`] = `
Array [
  Object {
    "borderRadius": 4,
    "paddingHorizontal": 8,
    "paddingVertical": 4,
  },
  Object {
    "backgroundColor": "#EE0612",
  },
]
`;
